File: truss-compiler/src/main/java/truss/compiler/parser/MemberAccessSelector.java
package truss.compiler.parser;

import truss.compiler.Span;
import truss.compiler.syntax.ExpressionSyntax;
import truss.compiler.syntax.MemberAccessExpressionSyntax;
import truss.compiler.syntax.SimpleNameSyntax;

class MemberAccessSelector implements Selector {
    private final SimpleNameSyntax memberName;
    private final Span span;

    public MemberAccessSelector(SimpleNameSyntax memberName, Span span) {
        this.memberName = memberName;
        this.span = span;
    }

    @Override
    public ExpressionSyntax build(ExpressionSyntax value) {
        return new MemberAccessExpressionSyntax(value, memberName, span);
    }
}

File: Truss.Compiler/Parser/MemberAccessSelector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Truss.Compiler.Syntax;

namespace Truss.Compiler.Parser {
    internal class MemberAccessSelector : ISelector {
        private readonly SimpleNameSyntax _memberName;
        private readonly Span _span;

        public MemberAccessSelector(SimpleNameSyntax memberName, Span span) {
            _memberName = memberName;
            _span = span;
        }

        public ExpressionSyntax Build(ExpressionSyntax value) {
            return new MemberAccessExpressionSyntax(value, _memberName, _span);
        }
    }
}
